internal/sql/repository/bulkUpdate/BulkUpdateRepository.go

```diff
@@ -0,0 +1,161 @@
+package bulkUpdate
+
+import (
+	"fmt"
+	"github.com/devtron-labs/devtron/internal/sql/repository/chartConfig"
+	"github.com/devtron-labs/devtron/internal/sql/repository/pipelineConfig"
+	"github.com/go-pg/pg"
+	"go.uber.org/zap"
+)
+
+type BulkUpdateReadme struct {
+	tableName struct{} `sql:"bulk_update_readme" pg:",discard_unknown_columns"`
+	Id        int      `sql:"id"`
+	Operation string   `sql:"operation"`
+	Script    string   `sql:"script"`
+	Readme    string   `sql:"readme"`
+}
+
+type BulkUpdateRepository interface {
+	BuildAppNameQuery(appNameIncludes []string, appNameExcludes []string) string
+	FindBulkUpdateReadme(operation string) (*BulkUpdateReadme, error)
+	FindBulkAppNameForGlobal(appNameIncludes []string, appNameExcludes []string) ([]*pipelineConfig.App, error)
+	FindBulkAppNameForEnv(appNameIncludes []string, appNameExcludes []string, envId int) ([]*pipelineConfig.App, error)
+	FindBulkChartsByAppNameSubstring(appNameIncludes []string, appNameExcludes []string) ([]*chartConfig.Chart, error)
+	FindBulkChartsEnvByAppNameSubstring(appNameIncludes []string, appNameExcludes []string, envId int) ([]*chartConfig.EnvConfigOverride, error)
+	BulkUpdateChartsValuesYamlAndGlobalOverrideById(final map[int]string) error
+	BulkUpdateChartsEnvYamlOverrideById(final map[int]string) error
+}
+
+func NewBulkUpdateRepository(dbConnection *pg.DB,
+	logger *zap.SugaredLogger) *BulkUpdateRepositoryImpl {
+	return &BulkUpdateRepositoryImpl{dbConnection: dbConnection,
+		logger: logger}
+}
+
+type BulkUpdateRepositoryImpl struct {
+	dbConnection *pg.DB
+	logger       *zap.SugaredLogger
+}
+
+func (repositoryImpl BulkUpdateRepositoryImpl) BuildAppNameQuery(appNameIncludes []string, appNameExcludes []string) string {
+	var appNameIncludesQuery string
```

Comment:
use https://stackoverflow.com/questions/4928054/postgresql-wildcard-like-for-any-of-a-list-of-words/4928415 for optimization